spec/feature/confirm_email_prompt_spec.rb

```diff
@@ -0,0 +1,103 @@
+RSpec.feature "Confirm email prompt" do
+  context "For a newly created account" do
+    let(:user) { FactoryBot.create(:user) }
+
+    scenario "Banner is present for user with unconfirmed email address" do
+      given_i_have_logged_in
+      when_i_navigate_to_home
+      then_i_see_the_confirmation_reminder_banner
+      when_i_navigate_to_manage
+      then_i_see_the_confirmation_reminder_banner
+      when_i_navigate_to_security
+      then_i_see_the_confirmation_reminder_banner
+    end
+
+    scenario "Resend form is prefilled on new user signup" do
+      given_i_have_logged_in
+      when_i_navigate_to_home
+      then_i_see_the_confirmation_reminder_banner
+      when_i_click_the_link_on_the_confirmation_banner
+      then_i_see_the_new_confirmaton_page_header
+      and_my_email_address_should_be_prefilled_in_the_form
+    end
+
+    scenario "Banner is not present once user has confirmed email address" do
+      given_i_have_logged_in
+      when_i_navigate_to_home
+      then_i_see_the_confirmation_reminder_banner
+      when_i_confirm_my_email_with_a_confirmation_link
+      when_i_navigate_to_home
+      then_i_do_not_see_the_confirmation_reminder_banner
+    end
+  end
+
+  context "For an existing user requesting to change their email address" do
+    let(:user) { FactoryBot.create(:user, :confirmed, :email_change_requested) }
+
+    scenario "Resend form is prefilled for changed email address" do
+      given_i_have_logged_in
+      when_i_navigate_to_home
+      then_i_see_the_confirmation_reminder_banner
+      when_i_click_the_link_on_the_confirmation_banner
+      then_i_see_the_new_confirmaton_page_header
+      and_my_unconfirmed_email_address_should_be_prefilled
+    end
+
+    scenario "Banner is not present once user has confirmed email address" do
+      given_i_have_logged_in
+      when_i_navigate_to_home
+      then_i_see_the_confirmation_reminder_banner
+      when_i_confirm_my_email_with_a_confirmation_link
+      when_i_navigate_to_home
+      then_i_do_not_see_the_confirmation_reminder_banner
+    end
+  end
+
+  def given_i_have_logged_in
+    visit "/"
```

Comment:
Is it possible to use the path helper method here?  It's likely the login route will change when we do the decoupling of login and registration, so this will reduce the number of changes needed.

Reply:
Great shout, will move that too and squash everything down